presto-cli/src/test/java/com/facebook/presto/cli/TestTemporaryFunctions.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.cli;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import org.testng.annotations.Test;
+
+import static com.facebook.presto.client.PrestoHeaders.PRESTO_ADDED_SESSION_FUNCTION;
+import static com.facebook.presto.client.PrestoHeaders.PRESTO_REMOVED_SESSION_FUNCTION;
+import static com.facebook.presto.client.PrestoHeaders.PRESTO_SESSION_FUNCTION;
+
+@Test
+public class TestTemporaryFunctions
+        extends AbstractCliTest
+{
+    @Test
+    public void testAddAndDropTempFunctions()
+            throws InterruptedException
+    {
+        server.enqueue(createMockResponse().addHeader(PRESTO_ADDED_SESSION_FUNCTION, "foo=foofunction"));
+        server.enqueue(createMockResponse().addHeader(PRESTO_ADDED_SESSION_FUNCTION, "bar=barfunction"));
+        server.enqueue(createMockResponse());
+        server.enqueue(createMockResponse().addHeader(PRESTO_REMOVED_SESSION_FUNCTION, "foo"));
+        server.enqueue(createMockResponse());
+        server.enqueue(createMockResponse().addHeader(PRESTO_REMOVED_SESSION_FUNCTION, "bar"));
+        server.enqueue(createMockResponse());
+
+        executeQueries(ImmutableList.of(
```

Comment:
Also test CREATE OR REPLACE

Reply:
We are not supporting REPLACE syntax yet